quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java

```diff
@@ -361,36 +362,42 @@ public static CommandLine createCommandLine(List<String> cliArgs) {
         return cmd;
     }
 
-    private static void addCommandOptions(List<String> cliArgs, CommandSpec command) {
+    private static void addCommandOptions(List<String> cliArgs, CommandLine command) {
         if (command != null) {
             boolean includeBuildTime = false;
             boolean includeRuntime = false;
 
-            if (Start.NAME.equals(command.name()) || StartDev.NAME.equals(command.name()) || Export.NAME.equals(command.name())) {
+            if (command.getCommand() instanceof AbstractCommand) {
+                AbstractCommand abstractStartCommand = command.getCommand();
+                includeRuntime = abstractStartCommand.includeRuntime();
+                includeBuildTime = abstractStartCommand.includeBuildTime();
+            }
+
+            if (!includeBuildTime && !includeRuntime) {
+                return;
+            } else if (includeRuntime && !includeBuildTime && (Start.NAME.equals(command.getCommandName())) || StartDev.NAME.equals(command.getCommandName())) {
```

Comment:
Thinking about this statement. We'll in the near future also provide the possibility to have Export with an optimized flag and probably even more commands in the future with that; this will be kinda messy. 

I was thinking something like this[1]. It's just a quick prototype, but it probably would be good to consider. For all 'Dependent' options, we just check the default action, and that's mainly related to the `--optimized` flag. But in the future, we could provide method sth like `evaluateDependentOptions(List<String> cliArgs){...}`, where we could specify when the dependent option should include also `build` process. 

However, it'd need to be properly documented and as these flags are used only on one place ATM, I don't see the high need to provide the additional logic. 

We can discuss this in the future; kinda out of scope of this PR.

[1] https://github.com/mabartos/keycloak/commit/31ae219a33ce109142f46a655acccbd6a6ce40d0

Reply:
@ahus1 Agree, thanks.